homeassistant/components/binary_sensor/zha.py

```diff
@@ -83,7 +87,17 @@ def cluster_command(self, tsn, command_id, args):
         if command_id == 0:
             self._state = args[0] & 3
             _LOGGER.debug("Updated alarm state: %s", self._state)
-            self.schedule_update_ha_state()
+            self.async_schedule_update_ha_state()
```

Comment:
Is `cluster_command` executed in an async context? Either this line or the call to `self.hass.add_job` 3 lines below is wrong.

Reply:
I think it is run in an async context. But that is based on my evidence that when I comment the `self.async_schedule_update_ha_state()` line my binary sensor no long updates the UI. Does that sound right?

Reply:
Added a new commit with the change 3 lines down. Confirmed the changes works fine.